### ISSUE TITLE
loginviauthinfo doesnt accept PWENCODED credentials

### ISSUE THREAD

**Issue by suktim** [sassoftware/pyviyatools#68]
we are working on a project that requires "Visual Analytics"  report backup on a daily, the only way to do that will be using one of the pyviyatool scripts.

We can easily create an automated strategy using the tool scripts in combination with Linux scripts and crontab schedule.

However, one thing that is a sticking point is credential exposure through the authinfo file. 

**ISSUE:** 

1. .authinfo file contains username and password 
2. The  user name and password are exposed 
3. Password exposure doesn't meet the security requirement 
4. loginviaauthinfo.py tool doesn't accept encoded passwords 

**Request:** 

1. Modify the tool to accept encoded password 
2. place encoded password in the .authinfo file 
3. pyviyatool accepts the entry  and then decodes the encoded password 
5.  It parses the decoded password through for authentication

**Urgency:** 
The customer is looking for a quick turnaround period, as it is integral for the project

**Comment by dwstern** [comment]
Hi Abiola,

Gerry may not be able to respond very quickly this week as he is teaching, so I hope you don't mind if I reply initially.

What makes you think the username and password are exposed? It's normal practice to secure the .authinfo file so that is has read-only permissions for the user, and no permissions for group or other: in other words, chmod 400 or -r-------- permissions. That way it is available only to the account that owns the .authinfo file. I don't think it is exposed if you do this.

Next, loginviauthinfo.py is a helper tool we wrote that ultimately calls the sas-admin CLI's built-in authentication command. If that command does not accept encoded passwords, our helper tool can't easily do so either. I don't believe it's a feature we'd likely implement in our wrapper/helper tool, it's a feature that would first need to be present in the sas-admin CLI, and we could then make sure we supported passing the encoded password on to that tool.

However, have you considered using Kerberos authentication for your project? That would be an existing, supported way of not storing a password anywhere in the clear, however well secured. See https://documentation.sas.com/?cdcId=calcdc&cdcVersion=3.5&docsetId=calcli&docsetTarget=n1e2dehluji7jon1gk69yggc6i28.htm&locale=en#n0npe77gdcwebsn1npqvifz4buu8, under "Use a Profile to Sign In".

Kind regards,

David
